crates/esthri/src/blocking.rs

```diff
@@ -151,7 +133,7 @@ pub async fn compute_etag(path: impl AsRef<Path>) -> Result<String> {
 }
 
 #[tokio::main]
-pub async fn delete<T>(s3: &T, bucket: impl AsRef<str>, keys: &[impl AsRef<str>]) -> Result<()>
+pub async fn delete<T>(s3: &T, bucket: &str, keys: Vec<String>) -> Result<()>
```

Comment:
Why `Vec<String>` over a borrowed type (here and elsewhere)?

Reply:
Seems like you might not want the keys to be consumed when this function is called

Reply:
@joelynch 

> Not really a refactor since you are changing the API of this crate in a bunch of places - we would need to update sitl to support these changes

@joelynch yes this is definitely too much for a single PR so don't worry about it, I will park it here for few days while working on a smaller PR then close it. There are multiple issues with it and hard to address them all at once, since we seems to have some diverging styles written over the years specifically on string handling. I am working on a writeup to address some of the patterns and hopefully you could assist me on this next week if time permits